pulsar-client-cpp/lib/ConsumerImpl.cc

```diff
@@ -925,9 +926,21 @@ void ConsumerImpl::acknowledgeAsync(const MessageId& msgId, ResultCallback callb
     doAcknowledgeIndividual(msgId, cb);
 }
 
+void ConsumerImpl::acknowledgeAsync(const MessageIdList& messageIdList, ResultCallback callback) {
+    ResultCallback cb = std::bind(&ConsumerImpl::statsAckCallback, shared_from_this(), std::placeholders::_1,
+                                  callback, proto::CommandAck_AckType_Individual, messageIdList.size());
+    // Currently not supported batch message id individual index ack.
+    this->ackGroupingTrackerPtr_->addAcknowledgeList(messageIdList);
+    for (const auto& messageId : messageIdList) {
+        this->unAckedMessageTrackerPtr_->remove(messageId);
+        this->batchAcknowledgementTracker_.deleteAckedMessage(messageId, proto::CommandAck::Individual);
+    }
```

Comment:
Yes, It is better to provide an accept `MessageIdList` method. I will provide it. 

But I want to know what is the race condition for the actual scenario. For `UnAckedMessageTracker`, Will there be the following scenario?

have a `MessageIdList`: {1, 2, 3, 4}

on acknowledge thread:

``` c++
for MessageIdList{
     this->unAckedMessageTrackerPtr_->remove(msgId);
}
```

and another thread to call: `this->unAckedMessageTrackerPtr_->add(msgId);`

Reply:
I add new `remove(MessageIdList)` method on `UnAckedMessageTracker`, Since `BatchAcknowledgementTracker` not support `Individual` ack, so, I remove invoke `this->batchAcknowledgementTracker_.deleteAckedMessage` code. PTAL. Thanks.